### PR TITLE
Feature/doi generation

### PR DESCRIPTION
**Pull request by BryceMarshall**
Added firebase functions 
Added buttons on data identification page to create/delete draft DOIs

### CODE REVIEW

**Comment by n-a-t-e** [comment]
Fixes #225

**Comment by n-a-t-e** [comment]
Now that the API call part works, looks like we need some metadata mapping to get some fields filled out in the draft DOI

Also, 
 - Clicking "Generate" should make the button switch to a spinner till API call is done
 - We should handle workflow for when the user changes the record after the draft is created, eg if the user changes the record's title, can they then click to update the DOI draft?

File: package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     "date-fns": "^2.0.0-beta.5",
     "file-saver": "^2.0.5",
     "firebase": "^7.24.0",
+    "firebase-admin": "^11.5.0",
+    "firebase-functions": "^4.2.1",
```
Comment (sorochak) [inline package.json:24]: I don't know if these two firebase packages should be dev dependencies?

File: src/components/Tabs/IdentificationTab.jsx [hunk 2]
```diff
@@ -55,675 +62,758 @@ const IdentificationTab = ({
...
               <En>
-                <p>Recommended title includes: What, Where, When.</p>
-                <p>
-                  Title should be precise enough so that the user will not have
-                  to open the dataset to understand its contents. Title should
-                  not have acronyms, special characters, or use specialized
-                  nomenclature. This will appear as the title that is shown for
-                  this dataset in the {regionInfo.catalogueTitle.en}.
-                </p>
+                What is the dataset title? Required in English and French.
```
Comment (sorochak) [inline src/components/Tabs/IdentificationTab.jsx:118]: I don't know why there's so much space below these tags, I don't see it in my editor.

Comment (sorochak) [reply src/components/Tabs/IdentificationTab.jsx:118]: it seems like my editor did some formatting and added a bunch of white space

Comment (n-a-t-e) [reply src/components/Tabs/IdentificationTab.jsx:118]: do you have the eslint plugin installed in vscode? it should read the .eslintrc file and match the formatting 🤞

File: package.json (hunk 1 above)
Comment (sorochak) [reply package.json:24]: I think these might have been needed in the project in order to run the emulator?

**Comment by sorochak** [comment]
TO DO: fix errors that occur when saving empty form.

**Review by n-a-t-e — APPROVED** [review]

**Pull request merged**